### PR TITLE
WIP Enabling Avgpool1d, pooling converters refactoring.

### PR DESCRIPTION
**Pull request by borisfom**
# Description

Enabling Avgpool1d, pooling converters refactoring.
Marked as WIP as it Includes/depends on  previous Conv1D PR. 
Also, I merged in MatMul PR to try passing my test case. No dependency on this one.
Need to add unit test, too.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [x ] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [x ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: core/conversion/converters/impl/pooling.cpp [hunk 1]
```diff
@@ -9,101 +10,146 @@ namespace converters {
...
+    new_layer = pooling_layer;
+  }
+	       
   new_layer->setName(util::node_info(n).c_str());
-  auto out_tensor = ctx->AssociateValueAndTensor(n->outputs()[0], new_layer->getOutput(0));
 
-  LOG_DEBUG("Output tensor shape: " << out_tensor->getDimensions());
+  if (shuffle_back) {
+    new_layer = util::unpadTensorDim(ctx, n, new_layer->getOutput(0), 3);
+    assert(new_layer);
```
Comment (peri044) [inline core/conversion/converters/impl/pooling.cpp:83]: use TRTORCH_CHECK here

File: cpp/trtorchc/main.cpp [hunk 2]
```diff
@@ -230,7 +230,7 @@ int main(int argc, char** argv) {
   }
 
   if (verbose) {
-    trtorch::logging::set_reportable_log_level(trtorch::logging::Level::kDEBUG);
+    trtorch::logging::set_reportable_log_level(trtorch::logging::Level::kGRAPH);
```
Comment (peri044) [inline cpp/trtorchc/main.cpp:233]: We can revert this back as kGRAPH prints too much info.

**Review by peri044 — CHANGES_REQUESTED** [review]
This PR includes the changes in this https://github.com/NVIDIA/TRTorch/pull/424. 
Either have it as a single PR or make them exclusive as there is a separate matmul PR code as well in this. For 1d conv, review comments are in #424 PR

File: WORKSPACE [hunk 3]
```diff
@@ -35,7 +35,7 @@ git_repository(
 # CUDA should be installed on the system locally
 new_local_repository(
     name = "cuda",
-    path = "/usr/local/cuda-11.1/",
+    path = "/usr/local/cuda/",
```
[the remark this answers is not in the document]
Comment (borisfom) [reply WORKSPACE:38]: On my box (cuda-11.2), cuda-11.1 directory only has nvrtc.

File: WORKSPACE [hunk 4]
```diff
@@ -69,21 +69,21 @@ http_archive(
...
 # or modify the urls to "file:///<PATH TO TARBALL>/<TARBALL NAME>.tar.gz
 
-http_archive(
-    name = "cudnn",
-    urls = ["https://developer.nvidia.com/compute/machine-learning/cudnn/secure/8.1.1.33/11.2_20210301/cudnn-11.2-linux-x64-v8.1.1.33.tgz",],
-    build_file = "@//third_party/cudnn/archive:BUILD",
-    sha256 = "98a8784e92862f20018d20c281b30d4a0cd951f93694f6433ccf4ae9c502ba6a",
-    strip_prefix = "cuda"
-)
+#http_archive(
```
[the remark this answers is not in the document]
Comment (borisfom) [reply WORKSPACE:72]: Again, on my setup, pip3_import commands simply fail. Same happens when I tried building in in Docker based on 11.02 With the update, it seems to work both on my setup and yours, and in Docker as well. I will put proposed WORKSPACE changes in a separate PR.

File: cpp/trtorchc/main.cpp (hunk 2 above)
Comment (borisfom) [reply cpp/trtorchc/main.cpp:233]: Thanks, I did not intend to include this debug change in PR.

**Comment by borisfom** [comment]
Superseded by https://github.com/NVIDIA/TRTorch/pull/435

**Pull request closed without merging**